File: decompiled/sources/com/hjq/toast/SafeHandler.java
package com.hjq.toast;

import android.os.Handler;
import android.os.Message;
import android.view.WindowManager;

final class SafeHandler extends Handler {
    private final Handler mHandler;

    SafeHandler(Handler handler) {
        this.mHandler = handler;
    }

    public void handleMessage(Message message) {
        try {
            this.mHandler.handleMessage(message);
        } catch (WindowManager.BadTokenException | IllegalStateException e) {
            e.printStackTrace();
        }
    }
}

File: decompiled/sources/com/hjq/toast/SafeHandler.kt
package com.hjq.toast

import android.os.Handler
import android.os.Message
import android.view.WindowManager

class SafeHandler(
    private val mHandler: Handler
) : Handler() {
    override fun handleMessage(message: Message) {
        try {
            mHandler.handleMessage(message)
        } catch (e: WindowManager.BadTokenException) {
            e.printStackTrace()
        } catch (e: IllegalStateException) {
            e.printStackTrace()
        }
    }
}
